packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -67,11 +67,9 @@ export default function({ types: t }) {
   function createObjectSpread(path, file, objRef) {
     const props = path.get("properties");
     const last = props[props.length - 1];
-    const restElement = last.node.type === "RestElement" && t.clone(last.node);
-
-    if (restElement) {
-      last.remove();
-    }
+    t.assertRestElement(last.node);
```

Comment:
Made this an assertion so that it fails immediately if our assumptions are violated.